ecs_deploy/ecs.py

```diff
@@ -663,6 +665,88 @@ def deploy(self, task_definition):
             raise EcsError(str(e))
 
 
+class DeployBlueGreenAction(EcsAction):
+    def __init__(self, *args, **kwargs):
+        super(DeployBlueGreenAction, self).__init__(*args, **kwargs)
+        self.cd_application_name = kwargs['cd_application_name']
+        self._cd_group_name = None
+        self._deployment_target_id = None
+        self._deployment_id = None
+        self._traffic_health_percentage = 90
+
+    @property
+    def deployment_id(self):
+        if not self._deployment_id:
+            raise EcsError('CodeDeploy deployment not defined')
+        return self._deployment_id
+
+    @property
+    def cd_group_name(self):
+        if not self._cd_group_name:
+            self._cd_group_name = self.client.codedeploy.list_deployment_groups(
+                applicationName=self.cd_application_name,
+            )['deploymentGroups'][0]
+        return self._cd_group_name
+
+    @property
+    def deployment_target_id(self):
+        while not self._deployment_target_id:
+            try:
+                self._deployment_target_id = self.client.codedeploy.list_deployment_targets(
+                    deploymentId=self.deployment_id
+                )['targetIds'][0]
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'DeploymentNotStartedException':
+                    time.sleep(1)
+                    continue
+                raise e
+        return self._deployment_target_id
+
+    def deploy(self, task_definition):
+        response = self.client.codedeploy.create_deployment(
```

Comment:
Actually this like makes the class to a "codedeploy-deploy" tool, rather then an "ecs-deploy" too. 
I'm still thinking, if this use-case might be a bit out of the scope of the project.